client/state/utils/README.md

```diff
@@ -0,0 +1,18 @@
+Utils
+========
+
+A module for those little helpers that need to interact woth the store.
+
+# Actions
+Used in combination with the Redux store instance `dispatch` function, actions can be used in manipulating the current global state.
+
+## uniqueId( prefix )
+Meant as a replacement for _ and lodash "uniqueId" function.
+Returns unique numeric prefixed by prefix and increments internal state, so the next uniqueId will be different.
```

Comment:
The rest of the `uniqueId` function looks like a regular `redux-thunk`-powered action creator, but the return value of the thunk itself is where things get funky. A thunk isn't expected to return anything at all, since it is just a delayed execution that will call `dispatch` whenever it decides.

See the [implementation of `redux-thunk`](https://github.com/gaearon/redux-thunk/blob/00f7fdb9b41e7445941692bc9c808f3769be22bd/src/index.js#L4): the output of `action( dispatch, getState )` — which is the invocation of the thunk — is never caught. I don't think it's our place to catch it.

tl;dr: a thunk should not be returning anything. I can see why this approach was followed, but by taking this shortcut — vs. going the regular Redux approach of accessing global state via `connect`+selector — we're fragmenting the way we all deal with data, and potentially opening holes through which things may fail at some point.